addons/web/models/res_users.py

```diff
@@ -9,7 +9,10 @@ class ResUsers(models.Model):
 
     @api.model
     def _name_search(self, name, domain=None, operator='ilike', limit=None, order=None):
-        user_ids = list(super()._name_search(name, domain, operator, limit, order))
+        user_ids = super()._name_search(name, domain, operator, limit, order)
```

Comment:
`user_ids` -> `user_query` or something else to clearly show that you handle a Query object and not a list of ids at this point.